torch/nn/modules/rnn.py

```diff
@@ -142,7 +142,7 @@ def forward(self, input, hx=None):
         is_packed = isinstance(input, PackedSequence)
         if is_packed:
             input, batch_sizes = input
-            max_batch_size = batch_sizes[0]
+            max_batch_size = batch_sizes.data[0]
```

Comment:
@colesbury Is this going to be affected by the Tensor/Variable merge?

Reply:
OK, it is going to be affected.

To make this code forwards compatible, you'll need to replace `batch_sizes.data[0]` with `int(batch_sizes[0])`.

Back story: in the old days, indexing into 1D tensors would give you a Python number back, but indexing into 1D variables would give you a (scalar) Variable. Since we are merging Variable and Tensor, we are changing the semantics of Tensor indexing to match Variable. This means that the code you wrote above will give you a PyTorch Scalar and will subsequently break when you try to use it later. Using an explicit conversion with `int()` solves the problem, and also solves you having to look at `data`.

Reply:
++